algorithms/src/snark/marlin/ahp/ahp.rs

```diff
@@ -237,19 +239,15 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
 
         // Lincheck sumcheck:
         let lincheck_time = start_timer!(|| "Lincheck");
-        let z_b_s = state
-            .circuit_specific_states
-            .iter()
-            .map(|(&circuit_id, circuit_state)| {
-                let z_b_i = (0..circuit_state.batch_size)
-                    .map(|i| {
-                        let z_b = witness_label(circuit_id, "z_b", i);
-                        LinearCombination::new(z_b.clone(), [(F::one(), z_b)])
-                    })
-                    .collect::<Vec<_>>();
-                (circuit_id, z_b_i)
-            })
-            .collect::<BTreeMap<_, _>>();
+        let mut z_b_s = BTreeMap::new();
+        for (circuit_id, circuit_state) in state.circuit_specific_states.iter() {
+            let mut z_b_i = Vec::with_capacity(usize::try_from(circuit_state.batch_size)?);
+            for i in 0..circuit_state.batch_size {
+                let z_b = witness_label(*circuit_id, "z_b", usize::try_from(i)?);
+                z_b_i.push(LinearCombination::new(z_b.clone(), [(F::one(), z_b)]));
+            }
+            z_b_s.insert(circuit_id, z_b_i);
+        }
```

Comment:
I think we can use `Itertools::try_collect` here, instead of rewriting as a for loop

Reply:
I don't want to return the Result, I have to unwrap the result in order to use z_b on line 246. I got it to compile if we return a `Result` anyway for all paths, but I don't think we'd be happy with this (the compiler asks for those type annotations: 

```
        let z_b_s = state
        .circuit_specific_states
        .iter()
        .map(|(&circuit_id, circuit_state)| {
            let z_b_i = (0..circuit_state.batch_size)
                .map(|i| {
                    let z_b = witness_label(circuit_id, "z_b", usize::try_from(i)?);
                    Ok::<LinearCombination<F>, std::num::TryFromIntError>(LinearCombination::new(z_b.clone(), [(F::one(), z_b)]))
                }).collect::<Result<Vec<_>,_>>();
            Ok::<(CircuitId, Vec<LinearCombination<F>>), std::num::TryFromIntError>((circuit_id, z_b_i?))
        })
        .collect::<Result<BTreeMap<_, _>,_>>()?;
```

Meanwhile, online fora on [related](https://stackoverflow.com/questions/62687455/alternatives-for-using-the-question-mark-operator-inside-a-map-function-closure) issues keep suggesting to just use a forloop.

Reply:
Got it down to 4916a3ca0 but it's still more code than the for-loop